umbraco/Umbraco-CMS
# RTE render macro as inline element option

### Prerequisites



### Description
I have noticed that a macro is "rendered" in a wrapper div when used in RTE, regardless of whether the actual macro outputs inline elements only. This means that if an editor would like to use an macro in the RTE as an inline macro, a div is always added to the RTE (even though this is temporary) but the RTE fixes the invalid markup by splitting the paragraph into three. The start P, the macro div, and the remaining P.

My PR gives the editor / developer the option of specifying per macro basis, if it should be rendered in RTE as an inline element, using a SPAN instead of a DIV

Test steps:
- on the Macro in the backoffice set "Render as inline element in rich text editor preview"
- Go to an RTE and add the macro
- Verify that it renders as span instead of DIV

unit tests has been added to MacroParserTests.cs

![image](https://user-images.githubusercontent.com/58024838/107225806-91562280-6a19-11eb-8df8-2402dc2b0b69.png)

![image](https://user-images.githubusercontent.com/58024838/107226160-0f1a2e00-6a1a-11eb-8501-5fa70a2a9fbe.png)

![image](https://user-images.githubusercontent.com/58024838/107226199-1a6d5980-6a1a-11eb-9e28-e9cc697ae59d.png)
